### PR TITLE
create parent for get data supplier so other file no need to import axios to get supplier data

### PR DESCRIPTION
**Pull request by Muh-Baharuddin**
@sidratul please review my code.

### CODE REVIEW

File: services/supplier/index.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import * as api from "services/api";
+import { QueryParamsType } from "src/@types/user";
+import { Data } from "./types";
+
+export const getAll = (params: QueryParamsType) => {
```
Comment (sidratul) [inline services/supplier/index.ts:5]: change to `getSuppliers`

File: services/supplier/types.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  id: string
+  nama: string
+  alamat: string
+  nomor_telepon: string
+}
+
+export interface Data {
+  data: Supplier[],
+  total: number;
+}
```
Comment (sidratul) [inline services/supplier/types.ts:11]: create as generic

File: src/containers/Supplier/Supplier.tsx [hunk 3]
```diff
@@ -54,14 +43,29 @@ export const DataSupplier = () => {
...
+    // })
+    
+    // const url = `${process.env.API_ENDPOINT}supplier`;
+    // axios.get<Data>(url, {
+    //   params: queryParams
+    // }).then((response) => {
+    //   setData(response.data);
+    // }).catch(() => {
+    //   toast.error("Maaf terjadi kesalahan pada server. Mohon coba kembali dalam beberapa saat.");
+    // })
```
Comment (sidratul) [inline src/containers/Supplier/Supplier.tsx:68]: remove

Comment (Muh-Baharuddin) [reply src/containers/Supplier/Supplier.tsx]: done remove comments

File: services/supplier/types.ts (hunk 2 above)
Comment (Muh-Baharuddin) [reply services/supplier/types.ts]: done,, create Data interface as Generic

File: services/supplier/index.ts (hunk 1 above)
Comment (Muh-Baharuddin) [reply services/supplier/index.ts]: done, rename getAll to getSuppliers

File: services/api.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import axios from 'axios';
+
+export const get = async <T>(url: string, params: Record<string, number | string | Date >): Promise<{
```
Comment (sidratul) [inline services/api.ts:3]: create type for `Record<string, number | string | Date >`

**Review by sidratul — APPROVED** [review]

File: services/api.ts (hunk 4 above)
Comment (Muh-Baharuddin) [reply services/api.ts]: done

**Pull request merged**